pkg/controllers/subnetbinding/subnetbinding_controller.go

```diff
@@ -208,36 +209,94 @@ func (r *Reconciler) listBindingMapIDsFromCRs(ctx context.Context) (sets.Set[str
 //     Subnet/SubnetSet readiness update will actively trigger a requeue event
 //  2. the associated Subnet is already used as a target Subnet in another SubnetConnectionBindingMap CR, or the target
 //     Subnet already has associated SubnetConnectionBindingMap CR. In this case, a retry error is returned.
+//  3. the target Subnet is a pre-created Subnet. In this case, a not-retry error is returned.
+//  4. the target Subnet and associated Subnet are not in the same vpc. In this case, not-retry error is returned.
 func (r *Reconciler) validateDependency(ctx context.Context, bindingMap *v1alpha1.SubnetConnectionBindingMap) (*model.VpcSubnet, []*model.VpcSubnet, *errorWithRetry) {
-	childSubnets, err := r.validateVpcSubnetsBySubnetCR(ctx, bindingMap.Namespace, bindingMap.Spec.SubnetName, false)
+	childSubnets, childSubnetCR, err := r.validateVpcSubnetsBySubnetCR(ctx, bindingMap.Namespace, bindingMap.Spec.SubnetName, false)
 	if err != nil {
 		return nil, nil, err
 	}
 	childSubnet := childSubnets[0]
 
 	if bindingMap.Spec.TargetSubnetName != "" {
-		parentSubnets, err := r.validateVpcSubnetsBySubnetCR(ctx, bindingMap.Namespace, bindingMap.Spec.TargetSubnetName, true)
+		parentSubnets, parentSubnetCR, err := r.validateVpcSubnetsBySubnetCR(ctx, bindingMap.Namespace, bindingMap.Spec.TargetSubnetName, true)
 		if err != nil {
 			return nil, nil, err
 		}
+		// check vpcName of child/parent Subnet
```

Comment:
Is it possible the vpcName are the same, but vpc paths are different, e.g., NSX orgs are different?
Why not directly compare the two VPC path since we already find the child/parent NSX VpcSubnet?

Reply:
Good point, updated the code to use Path directly. Thanks!